src/components/DragAndDrop/Provider/index.js

```diff
@@ -0,0 +1,60 @@
+import _ from 'underscore';
+import React, {useEffect, useRef} from 'react';
+import {View} from 'react-native';
+import {PortalHost} from '@gorhom/portal';
+import dragAndDropProviderPropTypes from './dragAndDropProviderPropTypes';
+import DNDUtils from '../Utils';
+import styles from '../../../styles/styles';
+import useDragAndDrop from '../../../hooks/useDragAndDrop';
+
+/**
+ * @param {Event} event – drag event
+ * @returns {Boolean}
+ */
+function shouldAcceptDrop(event) {
+    return _.some(event.dataTransfer.types, (type) => type === 'Files');
+}
+
+function DragAndDropProvider({children, dropZoneID, isDisabled = false}) {
+    const DragAndDropContext = DNDUtils.getDragAndDropContext(dropZoneID, true);
+    useEffect(
+        () => () => {
+            DNDUtils.deleteDragAndDropContext(dropZoneID);
+        },
+        [dropZoneID],
+    );
+
+    const dropZone = useRef(null);
+    const {isDraggingOver} = useDragAndDrop({
+        dropZone,
+        onDrop: (event) => {
+            DNDUtils.executeOnDropCallbacks(event, dropZoneID);
+        },
+        shouldAcceptDrop,
+        isDisabled,
+    });
+
+    return (
+        <DragAndDropContext.Provider value={{isDraggingOver}}>
+            <View
+                ref={(e) => (dropZone.current = e)}
+                style={[styles.flex1, styles.w100, styles.h100]}
```

Comment:
Is this considered inline styles? Or not since they are from the styles file?

Reply:
I would say no since they come from the styles file